ZQL crash when putting null

```
$ cat in.tzng 
#0:record[a:int32]
0:[-;]
12:04 ~/work/brim/zq(master)
$ zq -t "put b=null" in.tzng 
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x18 pc=0x13f07b1]

goroutine 1 [running]:
github.com/brimsec/zq/zngnative.ToNativeValue(0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0, ...)
        /Users/henridf/work/brim/zq/zngnative/native.go:19 +0x31
github.com/brimsec/zq/expr.compileNative(0x1a693c0, 0xc0003aaa80, 0x20, 0x179a0c0, 0xc00031f401)
        /Users/henridf/work/brim/zq/expr/expr.go:64 +0xc54
github.com/brimsec/zq/expr.CompileExpr(0x1a693c0, 0xc0003aaa80, 0x1, 0xc0003a0aa0, 0x50)
        /Users/henridf/work/brim/zq/expr/expr.go:42 +0x39
github.com/brimsec/zq/proc/put.New(0xc0003aaff0, 0x1a71960, 0xc0001b6aa0, 0xc0003aaab0, 0x8, 0x50, 0x17f1700, 0x203000)
        /Users/henridf/work/brim/zq/proc/put/put.go:63 +0xfc
github.com/brimsec/zq/proc/compiler.compileProc(0x0, 0x1a694a0, 0xc0003aaab0, 0xc0003aaff0, 0x1a71960, 0xc0001b6aa0, 0x203000, 0x203000, 0xc00031f650, 0x203000)
        /Users/henridf/work/brim/zq/proc/compiler/compiler.go:111 +0xeec
github.com/brimsec/zq/proc/compiler.Compile(0x0, 0x1a694a0, 0xc0003aaab0, 0xc0003aaff0, 0xc0003a7550, 0x1, 0x1, 0x0, 0x0, 0xc00005c800, ...)
        /Users/henridf/work/brim/zq/proc/compiler/compiler.go:189 +0x295
github.com/brimsec/zq/proc/compiler.compileSequential(0x0, 0xc0003a0750, 0x1, 0x1, 0xc0003aaff0, 0xc0003a7550, 0x1, 0x1, 0xc0001b6aa0, 0x1, ...)
        /Users/henridf/work/brim/zq/proc/compiler/compiler.go:131 +0x9a
github.com/brimsec/zq/proc/compiler.Compile(0x0, 0x1a69500, 0xc0003aafc0, 0xc0003aaff0, 0xc0003a7550, 0x1, 0x1, 0x1, 0x0, 0x7fffffffffffffff, ...)
        /Users/henridf/work/brim/zq/proc/compiler/compiler.go:196 +0x140
github.com/brimsec/zq/driver.compile(0x1a7a920, 0xc00037f180, 0x1a69500, 0xc0003aafc0, 0xc0001b6910, 0x1a71920, 0xc0003aaf60, 0x0, 0xc00019b740, 0x1, ...)
        /Users/henridf/work/brim/zq/driver/compile.go:73 +0x34a
github.com/brimsec/zq/driver.MultiRun(0x1a7aca0, 0xc0003aaf00, 0x1a7ac20, 0xc0003aaed0, 0x1a69500, 0xc0003aaae0, 0xc0001b6910, 0x1a71920, 0xc0003aaf60, 0x0, ...)
        /Users/henridf/work/brim/zq/driver/driver.go:32 +0x111
github.com/brimsec/zq/driver.Run(0x1a7aca0, 0xc0003aaf00, 0x1a7ac20, 0xc0003aaed0, 0x1a69500, 0xc0003aaae0, 0xc0001b6910, 0x1a69700, 0xc0003aae40, 0x0, ...)
        /Users/henridf/work/brim/zq/driver/driver.go:25 +0x26b
main.(*Command).Run(0xc000396000, 0xc0001a2020, 0x2, 0x2, 0x0, 0x0)
        /Users/henridf/work/brim/zq/cmd/zq/zq.go:157 +0x951
github.com/mccanne/charm.(*instance).run(0xc000191b40, 0xc0001a2010, 0x3, 0x3, 0x0, 0x0)
        /Users/henridf/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/instance.go:53 +0x28c
github.com/mccanne/charm.(*Spec).ExecRoot(0x1f7a0a0, 0xc0001a2010, 0x3, 0x3, 0xc00031ff78, 0x10077af, 0xc00008e058, 0x0)
        /Users/henridf/go/pkg/mod/github.com/mccanne/charm@v0.0.3-0.20191224190439-b05e1b7b1be3/charm.go:77 +0x96
main.main()
        /Users/henridf/work/brim/zq/cmd/zq/main.go:9 +0x78
```

--- a/proc/put/ztests/null.yaml
+++ b/proc/put/ztests/null.yaml
@@ -1,0 +1,9 @@
+zql: put b=null
+
+input: |
+  #0:record[a:int32]
+  0:[1;]
+
+output: |
+  #0:record[a:int32,b:null]
+  0:[1;-;]

--- a/zng/null.go
+++ b/zng/null.go
@@ -1,17 +1,18 @@
 package zng
 
 import (
-	"errors"
+	"fmt"
 
 	"github.com/brimsec/zq/zcode"
 )
 
-var ErrInstantiateNull = errors.New("cannot instantiate type null")
-
 type TypeOfNull struct{}
 
 func (t *TypeOfNull) Parse(in []byte) (zcode.Bytes, error) {
-	return nil, ErrInstantiateNull
+	if len(in) > 0 {
+		return nil, fmt.Errorf("cannot instantiate type null with nonempty value %q", in)
+	}
+	return nil, nil
 }
 
 func (t *TypeOfNull) ID() int {

--- a/zng/value.go
+++ b/zng/value.go
@@ -19,12 +19,9 @@ type Value struct {
 	Bytes zcode.Bytes
 }
 
-// Parse translates an Literal into a Value.
+// Parse translates an ast.Literal into a Value.
 // This currently supports only primitive literals.
 func Parse(v ast.Literal) (Value, error) {
-	if v.Type == "null" {
-		return Value{}, nil
-	}
 	t := LookupPrimitive(v.Type)
 	if t == nil {
 		return Value{}, fmt.Errorf("unsupported type %s in ast.Literal", v.Type)

--- a/ztests/suite/errors/null.yaml
+++ b/ztests/suite/errors/null.yaml
@@ -10,4 +10,4 @@ inputs:
 outputs:
   - name: stderr
     regexp: |
-      cannot instantiate type null
+      cannot instantiate type null with nonempty value "bleah"

